x/cdp/keeper/cdp.go

```diff
@@ -340,72 +341,73 @@ func (k Keeper) SetGovDenom(ctx sdk.Context, denom string) {
 }
 
 // ValidateCollateral validates that a collateral is valid for use in cdps
-func (k Keeper) ValidateCollateral(ctx sdk.Context, collateral sdk.Coins) sdk.Error {
+func (k Keeper) ValidateCollateral(ctx sdk.Context, collateral sdk.Coins) error {
 	if len(collateral) != 1 {
-		return types.ErrInvalidCollateralLength(k.codespace, len(collateral))
+		return sdkerrors.Wrapf(types.ErrInvalidCollateralLength, "%d", len(collateral))
 	}
 	_, found := k.GetCollateral(ctx, collateral[0].Denom)
 	if !found {
-		return types.ErrCollateralNotSupported(k.codespace, collateral[0].Denom)
+		return sdkerrors.Wrap(types.ErrCollateralNotSupported, collateral[0].Denom)
 	}
 	return nil
 }
 
 // ValidatePrincipalAdd validates that an asset is valid for use as debt when creating a new cdp
-func (k Keeper) ValidatePrincipalAdd(ctx sdk.Context, principal sdk.Coins) sdk.Error {
+func (k Keeper) ValidatePrincipalAdd(ctx sdk.Context, principal sdk.Coins) error {
 	for _, dc := range principal {
 		dp, found := k.GetDebtParam(ctx, dc.Denom)
 		if !found {
-			return types.ErrDebtNotSupported(k.codespace, dc.Denom)
+			return sdkerrors.Wrap(types.ErrDebtNotSupported, dc.Denom)
 		}
 		if dc.Amount.LT(dp.DebtFloor) {
-			return types.ErrBelowDebtFloor(k.codespace, sdk.NewCoins(dc), dp.DebtFloor)
+			return sdkerrors.Wrapf(types.ErrBelowDebtFloor, "proposed %s < minimum %s", sdk.NewCoins(dc), dp.DebtFloor)
```

Comment:
```suggestion
			return sdkerrors.Wrapf(types.ErrBelowDebtFloor, "proposed %s < minimum %s", sdk.Coins{dc}, dp.DebtFloor)
```
While we're changing the errors do you want to remove `NewCoins` from within error messages? If `dc` is invalid in some way (such < 0) then `NewCoins` will panic. The simulator caught this in auction.

Reply:
I'd propose instead: `"proposed %s < minimum %s", dc, dp.DebtFloor)` since it's the same

Reply:
ah yeah, thats nicer